leakcanary-android-utils/src/main/java/leakcanary/DumpWrapper.kt

```diff
@@ -0,0 +1,42 @@
+package leakcanary
+
+import android.os.Build
+import android.os.Debug
+import android.util.Log
+import leakcanary.internal.checkNotMainThread
+import java.io.IOException
+import java.lang.Exception
+
+object DumpWrapper {
+  private const val TAG = "DumpWrapper"
+  private const val ANDROID_R = Build.VERSION_CODES.Q + 1
+  private val VERSION_MATCH = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && Build
+    .VERSION.SDK_INT <= ANDROID_R
+
+  @Throws(IOException::class)
+  fun dumpHprofData(fileName: String) {
+    if (isSupportFastDump()) {
+      checkNotMainThread()
```

Comment:
Why is not main thread important?

Reply:
ok. The trick is, this is currently exposed as an API on the LeakCanary object so someone might call it from the main thread. I'm not sure what to do, if we should enforce that at the callsite, switch off to a background thread, or just not use fast dump if called from the main thread.

Reply:
I think fallback in main thread called situation is better. If someone does not realize that fast dump should be called in a background thread, it is hardly for him to use fast dump correctly.
Switch off to a background thread automatically would make someone confused.

Reply:
Interesting. So what you're saying is that with "fast dump" the dumping isn't faster, but it's non blocking which means the parent process can keep running while the dump is happening in a child process, so it doesn't make sense to wait for that on the main thread and block. And in that case we might as well use the default API. Ok I like it.